purchases/src/main/kotlin/com/revenuecat/purchases/google/usecase/QueryPurchasesUseCase.kt

```diff
@@ -45,142 +26,58 @@ internal class QueryPurchasesUseCase(
-            queryParams = queryInAppsPurchasesParams,
-            listener = { unconsumedInAppsResult, unconsumedInAppsPurchases ->
-                processResult(
-                    unconsumedInAppsResult,
-                    unconsumedInAppsPurchases.toMapOfGooglePurchaseWrapper(BillingClient.ProductType.INAPP),
-                    onQueryInAppsSuccess,
-                    onQueryInAppsError,
-                )
-            },
-        )
+        QueryPurchasesByTypeUseCase(
+            useCaseParams = QueryPurchasesByTypeUseCaseParams(
+                dateProvider = useCaseParams.dateProvider,
+                diagnosticsTrackerIfEnabled = useCaseParams.diagnosticsTrackerIfEnabled,
+                appInBackground = useCaseParams.appInBackground,
+                productType = ProductType.INAPP,
+            ),
+            onSuccess = onQueryInAppsSuccess,
+            onError = onQueryInAppsError,
+            withConnectedClient = withConnectedClient,
+            executeRequestOnUIThread = executeRequestOnUIThread,
+        ).run()
     }
 
     private fun querySubscriptions(
-        billingClient: BillingClient,
         onQuerySubscriptionsSuccess: (Map<String, StoreTransaction>) -> Unit,
-        onQuerySubscriptionsError: (BillingResult) -> Unit,
+        onQuerySubscriptionsError: (PurchasesError) -> Unit,
     ) {
-        val querySubsPurchasesParams = BillingClient.ProductType.SUBS.buildQueryPurchasesParams()
-        if (querySubsPurchasesParams == null) {
-            val purchasesError = PurchasesError(
-                PurchasesErrorCode.PurchaseInvalidError,
-                PurchaseStrings.INVALID_PRODUCT_TYPE.format("queryPurchases"),
-            )
-            onError(purchasesError)
-            return
-        }
-        queryPurchasesAsyncWithTrackingEnsuringOneResponse(
-            billingClient,
-            BillingClient.ProductType.SUBS,
-            querySubsPurchasesParams,
-        ) { activeSubsResult, activeSubsPurchases ->
-            processResult(
-                activeSubsResult,
-                activeSubsPurchases.toMapOfGooglePurchaseWrapper(BillingClient.ProductType.SUBS),
-                onQuerySubscriptionsSuccess,
-                onQuerySubscriptionsError,
-            )
-        }
+        QueryPurchasesByTypeUseCase(
+            useCaseParams = QueryPurchasesByTypeUseCaseParams(
+                dateProvider = useCaseParams.dateProvider,
+                diagnosticsTrackerIfEnabled = useCaseParams.diagnosticsTrackerIfEnabled,
+                appInBackground = useCaseParams.appInBackground,
+                productType = ProductType.SUBS,
+            ),
+            onSuccess = onQuerySubscriptionsSuccess,
+            onError = onQuerySubscriptionsError,
+            withConnectedClient = withConnectedClient,
+            executeRequestOnUIThread = executeRequestOnUIThread,
+        ).run()
     }
 
     override fun executeAsync() {
```

Comment:
I mentioned it in the description of the PR, but this change now has the consequence of posting to the main thread twice, once for SUBS and once for INAPPS. I don't think it's a bad change, but maybe we want to consider removing this UseCase and moving its "logic" (calling the `QueryPurchasesByTypeUseCase` twice) to the BillingWrapper

Reply:
Hmm I guess a few things to be aware of are:
- There is a chance of this request to get delayed if something else is added to the `serviceRequests` in between both calls. Probably not a big deal
- If the billing client disconnects in between, we might need to reconnect. As well, probably not a big deal. 

One thing to be aware of though, if we try to execute the request with a delay, the delay would be applied twice, once for querying subs and once for inapps. I don't think we are adding a delay to these right now, so we should be ok I think.

Reply:
> If the billing client disconnects in between, we might need to reconnect. As well, probably not a big deal.

The good thing about this is we already have the result of one of the calls, so only one of the calls needs to be retried. This can also happen right now btw, and we would have to retry both

> I don't think we are adding a delay to these right now, so we should be ok I think.

Yes... We are not applying backoff here so I think it should be fine yes.

Reply:
Oh actually, we would backoff if the app is in background and we get a ServiceUnavailable

Reply:
Ohhh right... I guess in that case this could get delayed quite a bit.

Then again, thinking about it, if this is in the background, it might be ok? As long as, once we come back to foreground, we don't have to wait the whole time (so we can enqueue multiple instances of this request and it's not an issue).